sopel/db.py

```diff
@@ -390,9 +400,47 @@ def merge_nick_groups(self, first_nick, second_nick):
 
     # CHANNEL FUNCTIONS
 
+    def get_channel_slug(self, chan):
+        """Return the case-normalized representation of ``channel``.
+
+        :param str channel: the channel name to normalize, with prefix
+                            (required)
+        :return str: the case-normalized channel name (or "slug"
+                     representation)
+
+        This is useful to make sure that a channel name is stored consistently
+        in both the bot's own database and third-party plugins'
+        databases/files, without regard for variation in case between
+        different clients and/or servers on the network.
+        """
+        chan = Identifier(chan)
+        slug = chan.lower()
+        session = self.ssession()
+        try:
+            count = session.query(ChannelValues) \
+                .filter(ChannelValues.channel == slug) \
+                .count()
+
+            if count == 0:
```

Comment:
I think a `# TODO: remove for Sopel 8` might be useful here, because I don't think we should handle this case this far in the future.

Reply:
I'd rather a follow-up issue, or adding a task to #1389.

Honestly, later in 7.x, I think we should just add a database field somewhere that stores a version number we can use to run migrations like this all at once, if needed. Removing these migration checks without first making sure that _all_ of a Sopel instance's values have been fixed could lead to some really bad debugging times—probably involving one or both of us scratching our heads, thinking "but the code doesn't work that way!" while an innocent user anxiously waits for help. 😅

Reply:
(Alternatively, we can pause this change until 7.1, and do a more comprehensive "all at once" migration strategy in the first place.)